packages/@uppy/image-generator/src/index.tsx

```diff
@@ -0,0 +1,288 @@
+  MinimalRequiredUppyFile,
+  UIPluginOptions,
+  UppyEventMap,
+} from '@uppy/core'
+import Uppy, { UIPlugin } from '@uppy/core'
+import { FilterInput, SearchView } from '@uppy/provider-views'
+import type { AssemblyOptions, AssemblyResult } from '@uppy/transloadit'
+import Transloadit from '@uppy/transloadit'
+import locale from './locale.js'
+
+export interface ImageGeneratorOptions extends UIPluginOptions {
+  assemblyOptions: (prompt: string) => Promise<AssemblyOptions>
+}
+
+interface PluginState extends Record<string, unknown> {
+  prompt: string
+  results: AssemblyResult[]
+  checkedResultIds: Set<AssemblyResult['id']>
+  loading: boolean
+  loadingMessageIndex: number
+  firstRun: boolean
+}
+
+const defaultState = {
+  prompt: '',
+  results: [],
+  checkedResultIds: new Set(),
+  loading: false,
+  loadingMessageIndex: 0,
+  firstRun: true,
+} satisfies PluginState
+
+const LOADING_MESSAGES = [
+  'generating1',
+  'generating2',
+  'generating3',
+  'generating4',
+  'generating5',
+] as const
+
+export default class ImageGenerator<
+  M extends Meta,
+  B extends Body,
+> extends UIPlugin<ImageGeneratorOptions, M, B, PluginState> {
+  private loadingInterval: ReturnType<typeof setInterval> | null = null
+
+  constructor(uppy: Uppy<M, B>, opts: ImageGeneratorOptions) {
+    super(uppy, opts)
+
+    this.id = this.opts.id || 'ImageGenerator'
+    this.title = 'Image Generator'
+    this.type = 'acquirer'
+
+    this.defaultLocale = locale
+
+    this.setPluginState(defaultState)
+
+    this.i18nInit()
+  }
+
+  install(): void {
+    const { target } = this.opts
+    if (target) {
+      this.mount(target, this)
+    }
+  }
+
+  uninstall(): void {
+    this.clearLoadingInterval()
+    this.unmount()
+  }
+
+  private clearLoadingInterval(): void {
+    if (this.loadingInterval) {
+      clearInterval(this.loadingInterval)
+      this.loadingInterval = null
+    }
+  }
+
+  private startLoadingAnimation(): void {
+    this.clearLoadingInterval()
+    this.loadingInterval = setInterval(() => {
+      const { loadingMessageIndex } = this.getPluginState()
+      const nextIndex = (loadingMessageIndex + 1) % LOADING_MESSAGES.length
+      this.setPluginState({ loadingMessageIndex: nextIndex })
+    }, 4000)
+  }
+
+  search = async () => {
+    const assemblyOptions = await this.opts.assemblyOptions(
+      this.getPluginState().prompt,
+    )
+    const localUppy = new Uppy<M, B>().use(Transloadit, {
+      waitForEncoding: true,
+      assemblyOptions,
+    })
+    const onResult: UppyEventMap<M, B>['transloadit:result'] = (
+      stepName,
+      result,
+    ) => {
+      const { results } = this.getPluginState()
+      this.setPluginState({ results: [...results, result], firstRun: false })
+    }
+    localUppy.on('transloadit:result', onResult)
+
+    // @ts-expect-error not typed because we do not depend on @uppy/dashboard
+    this.uppy.once('dashboard:close-panel', () => {
+      localUppy.off('transloadit:result', onResult)
```

Comment:
should we not also clear interval here?
```suggestion
      this.clearLoadingInterval()
      localUppy.off('transloadit:result', onResult)
```